contracts/protocol/libs/FundsLib.sol

```diff
@@ -202,29 +201,134 @@ library FundsLib {
             }
         }
 
-        // Store payoffs to availablefunds and notify the external observers
         address exchangeToken = offer.exchangeToken;
-        uint256 sellerId = offer.sellerId;
-        uint256 buyerId = exchange.buyerId;
+
+        // Original seller and last buyer are done
+        // Release funds to intermediate sellers (if they exist)
+        // and add the protocol fee to the total
+        {
+            (uint256 sequentialProtocolFee, uint256 sequentialRoyalties) = releaseFundsToIntermediateSellers(
+                _exchangeId,
+                exchange.state,
+                price,
+                exchangeToken
+            );
+            sellerPayoff += sequentialRoyalties; // revisit this
```

Comment:
👀

Reply:
I guess that since tests pass, no further revision is needed :grin: